### PR TITLE
Script to populate parameter names/description i18n strings from documentation md files [finishes #93242318]

### PR DESCRIPTION
**Pull request by bcjordan**
Includes first alphabetization and import of parameter information

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
lgtm, tho im not ruby expert

File: dashboard/scripts/archive/block_docs_to_i18n [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  sorted_data_hash = Hash[modified_data_hash.sort]
+
+  File.write(json_file, JSON.pretty_generate(sorted_data_hash) + "\n")
+end
+
+doc_files = Dir.glob("#{CDO_ROOT}pegasus/**/applab/docs/*.md").select { |e| File.file? e }
+puts "Processing #{doc_files.count} doc files..."
+function_infos = parse_function_docs(doc_files)
+puts "Updating i18n lines..."
+update_docs_i18n_file(function_infos, "#{CDO_ROOT}apps/i18n/common/en_us.json")
```
Comment (laurelfan) [inline dashboard/scripts/archive/block_docs_to_i18n]: maybe consider catching exceptions thrown by parse_function_docs and printing an error message with the filename?

File: dashboard/scripts/archive/block_docs_to_i18n [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  new_strings = {}
+
+  functions.each do |f|
+    f.params.each_with_index do |param, i|
+      new_strings["dropletBlock_#{f.name}_param#{i}"] = param.name
+      new_strings["dropletBlock_#{f.name}_param#{i}_description"] = param.description
+    end
+  end
+
+  json_file = i18n_filename
```
Comment (laurelfan) [inline dashboard/scripts/archive/block_docs_to_i18n]: doesn't look like this local var is doing much

**Comment by laurelfan** [comment]
is this going to be run 

I wouldn't worry about the travis failure, I'm going to fix it

**Comment by bcjordan** [comment]
@laurelfan this will be run manually (and fairly rarely) by Sarah to batch-update params from documents. There's also intended to be a manual inspection before committing

File: dashboard/scripts/archive/block_docs_to_i18n (hunk 2 above)
Comment (bcjordan) [inline dashboard/scripts/archive/block_docs_to_i18n]: Oops, silly remnant of a refactor, removing

**Comment by laurelfan** [comment]
ok, cool, looks good to me. sounds like manual inspection of the output is your testing... the detailed logging I suggested is less of an issue when it's not going to be automatically run, if Sarah gets a failure that she can't figure out she can just bother you ;)

**Comment by bcjordan** [comment]
I liked the file-reporting logging idea, added. e.g. output:

```
Processing 105 doc files...
Error processing file ../../../pegasus/sites.v3/code.org/public/applab/docs/moveForward.md:
undefined method `strip' for nil:NilClass
["/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:34:in `block in parse_function_doc'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:22:in `each'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:22:in `parse_function_doc'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:49:in `block in parse_function_docs'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:47:in `each'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:47:in `parse_function_docs'", "/Users/brian/code/code-dot-org/dashboard/scripts/archive/block_docs_to_i18n:79:in `<top (required)>'", "-e:1:in `load'", "-e:1:in `<main>'"]
Updating i18n lines...
⊙‿⊙

Process finished with exit code 0
```

**Comment by laurelfan** [comment]
cool ⊙‿⊙

**Pull request merged**